sql/src/main/java/io/crate/analyze/relations/QueriedRelation.java

```diff
@@ -26,4 +26,9 @@
 public interface QueriedRelation extends AnalyzedRelation {
 
     QuerySpec querySpec();
+
+    // relation Id used to distinguish between relations in the Fetch phase (union)
+    byte relationId();
+
+    void relationId(byte relationId);
```

Comment:
I don't think that having gaps in the relationIds is a problem.

Imho any objects that are "half" constructed and not fully initialized are always hard to deal with.
Especially if no-one is there to assert that they have been properly post-processed.

If we do it this way we at least have to add an assertion in the planner that checks that the relationIds are not duplicated.